src/main/java/soconnect/model/person/search/ContactMightBeRelevantPredicate.java

```diff
@@ -1,43 +1,70 @@
      */
-    public ContactMightBeRelevantPredicate(List<String> keywords) {
-        this.keywords = keywords;
+    public ContactMightBeRelevantPredicate(ArgumentMultimap argMultimap) {
+        this.argMultimap = argMultimap;
     }
 
     @Override
     public boolean test(Person person) {
-        boolean isNameContained = keywords.stream()
-                .allMatch(keyword -> StringUtil.containsSomeKeywordsIgnoreCase(
-                        person.getName().fullName, keyword
-                ));
-        boolean isAddressContained = keywords.stream()
-                .allMatch(keyword -> StringUtil.containsSomeKeywordsIgnoreCase(
-                        person.getAddress().value, keyword));
-        boolean isEmailContained = keywords.stream()
-                .allMatch(keyword -> StringUtil.containsSomeKeywordsIgnoreCase(
-                        person.getEmail().value, keyword));
-        boolean isPhoneContained = keywords.stream()
-                .allMatch(keyword -> StringUtil.containsSomeKeywordsIgnoreCase(
-                        person.getPhone().value, keyword));
-        boolean isTagContained = keywords.stream()
-                .allMatch(keyword -> person.getTags().stream()
-                        .anyMatch(tag -> StringUtil.containsSomeKeywordsIgnoreCase(tag.tagName, keyword)));
+        for (Prefix prefix : argMultimap.getAllPrefixes()) {
+            SearchPrefix.SearchPrefixCommand prefixCommand = convertPrefixToEnumType(prefix);
+            List<String> keywords = argMultimap.getAllValues(prefix);
+            switch (prefixCommand) {
+            case NAME:
+                isNameContained = keywords.stream()
+                        .anyMatch(keyword -> StringUtil
+                                .containsSomeKeywordsIgnoreCase(person.getName().fullName, keyword));
+                break;
+            case ADDRESS:
+                isAddressContained = keywords.stream()
+                        .anyMatch(keyword -> StringUtil
+                                .containsSomeKeywordsIgnoreCase(person.getAddress().value, keyword));
+                break;
+            case EMAIL:
+                isEmailContained = keywords.stream()
+                        .anyMatch(keyword -> StringUtil
+                                .containsSomeKeywordsIgnoreCase(person.getEmail().value, keyword));
+                break;
+            case PHONE:
+                isPhoneContained = keywords.stream()
+                        .anyMatch(keyword -> StringUtil
+                                .containsSomeKeywordsIgnoreCase(person.getPhone().value, keyword));
+                break;
+            case TAG:
+                isTagContained = keywords.stream()
+                        .anyMatch(keyword -> person.getTags().stream()
+                                .anyMatch(tag -> StringUtil
+                                        .containsSomeKeywordsIgnoreCase(tag.tagName, keyword)));
```

Comment:
This method exceeds 30LOC, you can shorten it by abstract out each of the 5 cases above.
E.g. isNameContained = containsNameKeyword(keywords);

Reply:
Nice catch, thank you!